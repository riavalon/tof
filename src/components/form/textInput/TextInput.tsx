import React, { ChangeEvent } from 'react'

import './TextInput.styles.scss'

export type textInputProps = {
  id: string;
  label: string;
  type?: string;
  placeholder?: string;
  className?: string;
  handleChange: (evt: ChangeEvent) => void,
}

const TextInput = ({
  id,
  type = 'text',
  label,
  placeholder,
  className,
  handleChange,
}: textInputProps) => (
  <div className={`tof-form-control${className ? ' ' + className : ''}`}>
    <label htmlFor={id}>{label}</label>
    <input
      type={type}
      onChange={(event: ChangeEvent) => handleChange(event)}
      id={id}
      placeholder={placeholder}
      data-testid={id}
    />
  </div>
)

export default TextInput