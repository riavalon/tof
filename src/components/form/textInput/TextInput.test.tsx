import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import TextInput, { textInputProps } from './TextInput'

describe('Form => Text Input', () => {
  let mockProps: textInputProps;

  beforeEach(() => {
    mockProps = {
      label: 'test label',
      type: 'text',
      handleChange: jest.fn(),
      id: 'test-id',
    }
  })

  it('should render with test id', () => {
    const { queryByTestId, baseElement } = render(<TextInput {...mockProps} />)
    expect(queryByTestId(/test-id/i)).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })

  it('should call provided function in props on change', () => {
    const { getByTestId } = render(<TextInput {...mockProps} />)
    const input = getByTestId(/test-id/i)
    fireEvent.change(input, { target: { value: 'testing' } })
    expect(mockProps.handleChange).toHaveBeenCalled()
  })
})