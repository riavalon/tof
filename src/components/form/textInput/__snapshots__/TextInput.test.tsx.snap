// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form => Text Input should render with test id 1`] = `
<body>
  <div>
    <div
      class="tof-form-control"
    >
      <label
        for="test-id"
      >
        test label
      </label>
      <input
        data-testid="test-id"
        id="test-id"
        type="text"
      />
    </div>
  </div>
</body>
`;
